### PR TITLE
Use container registry from Quay

### PR DESCRIPTION
**Pull request by fgimenez**
According to this docker proxy log https://gist.github.com/fgimenez/35c4fae7d994b55875b7ba6582311773 the registry image is the only one that we are pulling from docker hub on kubevirt's `make cluster-up && make cluster-sync`

/cc @dhiller @dankenigsberg 

Signed-off-by: Federico Gimenez <fgimenez@redhat.com>

### CODE REVIEW

File: cluster-provision/gocli/cmd/utils/images.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ const (
 	// NFSGaneshaImage contains the reference to NFS docker image
 	NFSGaneshaImage = "docker.io/janeczku/nfs-ganesha@sha256:17fe1813fd20d9fdfa497a26c8a2e39dd49748cd39dbb0559df7627d9bcf4c53"
```
Comment (dankenigsberg) [inline cluster-provision/gocli/cmd/utils/images.go:5]: this seems to be used in `cluster-provision`. Does it have a quay.io-stored alternative?
(should be addressed in another PR)

BTW,
```
$ git grep docker.io|wc -l
70
```
we should probably aim to drive this down

Comment (dhiller) [reply cluster-provision/gocli/cmd/utils/images.go:5]: I think most of the images that are resulting from this grep should be already pre-pulled during cluster-provisioning and then present in the kubevirtci image. Example for 1.21:
```
$ cd kubevirtci/cluster-provision/k8s/1.21
$ ./fetch-images.sh .
ceph/ceph:v15
docker.io/calico/cni:v3.18.0
docker.io/calico/kube-controllers:v3.18.0
docker.io/calico/node:v3.18.0
docker.io/calico/pod2daemon-flexvol:v3.18.0
fluent/fluentd-kubernetes-daemonset:v1.2-debian-syslog
grafana/grafana:7.5.4
k8s.gcr.io/kube-state-metrics/kube-state-metrics:v2.0.0
k8s.gcr.io/sig-storage/csi-attacher:v3.0.0
k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.0.1
k8s.gcr.io/sig-storage/csi-provisioner:v2.0.0
k8s.gcr.io/sig-storage/csi-resizer:v1.0.0
k8s.gcr.io/sig-storage/csi-snapshotter:v3.0.0
k8s.gcr.io/sig-storage/snapshot-controller:v4.0.0
quay.io/brancz/kube-rbac-proxy:v0.8.0
quay.io/cephcsi/cephcsi:v3.2.0
quay.io/external_storage/local-volume-provisioner:v2.3.2
quay.io/kubevirt/bridge-marker@sha256:8e9de03dbf6f35eb88af94e3d93da66d926d002c1574619f63186f1e88bf3c52
quay.io/kubevirt/cdi-apiserver:v1.36.0
quay.io/kubevirt/cdi-cloner:v1.36.0
quay.io/kubevirt/cdi-controller:v1.36.0
quay.io/kubevirt/cdi-importer:v1.36.0
quay.io/kubevirt/cdi-operator:v1.36.0
quay.io/kubevirt/cdi-uploadproxy:v1.36.0
quay.io/kubevirt/cdi-uploadserver:v1.36.0
quay.io/kubevirt/cluster-network-addon-multus@sha256:32867c73cda4d605651b898dc85fea67d93191c47f27e1ad9e9f2b9041c518de
quay.io/kubevirt/cluster-network-addons-operator:v0.58.0
quay.io/kubevirt/cni-default-plugins@sha256:828df9e296db6f2772e69671189c68b69c258fdc9c84b4cc08b16866f69c162d
quay.io/kubevirt/kubemacpool@sha256:4e1a54bf29456fd020cd8cbe9ea4b36e84f7f29eb342a469f6fba72da4b450a1
quay.io/kubevirt/macvtap-cni@sha256:e34cc796dee2e300f866d6f5b563361253ce89226eaf9eb0c3bc792f5481b8df
quay.io/kubevirt/ovs-cni-marker@sha256:33671de777943e404d1cf980727b9c3cfb5b73dba1624614ab8a179bf02f6473
quay.io/kubevirt/ovs-cni-plugin@sha256:bb6acd2712482746da774af365afba2632beeacdb085e5266d527e3b08047466
quay.io/nmstate/kubernetes-nmstate-handler@sha256:18041db3cb27c673121e798b090e54cc42f6591f877ff599fa09f55c6d2388fe
quay.io/prometheus/alertmanager:v0.21.0
quay.io/prometheus/node-exporter:v1.1.2
quay.io/prometheus-operator/prometheus-operator:v0.47.0
quay.io/prometheus/prometheus:v2.26.0
rook/ceph:v1.5.8
$ cat extra-pre-pull-images
fluent/fluentd:v1.2-debian
docker.io/grafana/grafana:7.5.4
docker.io/istio/install-cni:1.10.0
docker.io/istio/operator:1.10.0
docker.io/istio/pilot:1.10.0
docker.io/istio/proxyv2:1.10.0
quay.io/prometheus-operator/prometheus-config-reloader:v0.47.0
```

Comment (dhiller) [reply cluster-provision/gocli/cmd/utils/images.go:5]: Correct me if I am wrong, @fgimenez but if we would add the container image proxy setup to the kubevirtci builds like we are doing with kubevirt, we would get rid of a lot of the docker rate limiting we are facing on kubevirtci builds, wouldn't we?

**Review by dhiller — COMMENTED** [review]
/approve
/lgtm
/hold

**Comment by dhiller** [comment]
Waiting for discussion to resolve.

File: cluster-provision/gocli/cmd/utils/images.go (hunk 1 above)
Comment (fgimenez) [reply cluster-provision/gocli/cmd/utils/images.go:5]: @dhiller sorry for the late response, 

You are totally right, looking at the job definitions here https://github.com/kubevirt/project-infra/blob/main/github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-presubmits.yaml we have set the mirror preset but we are using an image that is not taking advantage of it. We would need to modify quay.io/kubevirtci/golang:v20210316-d295087 to use our proxy, I can take care of that

Comment (fgimenez) [reply cluster-provision/gocli/cmd/utils/images.go:5]: > We would need to modify quay.io/kubevirtci/golang:v20210316-d295087 to use our proxy, I can take care of that

@dhiller feel free to work on it, let me know if you need any help :)
